### PR TITLE
Start enhancements

### PR DESCRIPTION
**Pull request by nzoschke**
This patch:

* vendors the go-dockerclient and uses it to introspect conflicting ports on start
* blocks to cleanly handle an interrupt signal

This comes with new tests for expectations around build and run output.

* makes the exec and signal wait functions overridable for easier testing
* tags images in alphabetical order for easier testing
* adds additional alphabetical ordering to the runOrder for easier testing
* fixes tests on Travis

### CODE REVIEW

**Comment by nzoschke** [comment]
I was expecting to do signal propagation but on testing the simple signal handler is working very well, gracefully terminating the children processes and containers.

**Comment by nzoschke** [comment]
I noticed some edge cases where runOrder will not be correct, namely around links with links. But we can fix that later if it becomes a serious problem.

File: docker/docker.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package docker
+
+import (
+	"fmt"
+	"os"
+
+	docker "github.com/convox/cli/Godeps/_workspace/src/github.com/fsouza/go-dockerclient"
+)
+
+func PortsUsed() ([]int64, error) {
```
Comment (ddollar) [inline docker/docker.go]: For this can we check at the OS level instead of just Docker? Something else on the machine might have the port open.

Comment (nzoschke) [inline docker/docker.go]: Good idea. Then I can remove the docker client and not worry about expectations on how to connect to docker

**Pull request merged**